code/src/LeafletProvider/Features/Directions.ts

```diff
@@ -0,0 +1,417 @@
+
+        /** Handler for the routesfound event from Leaflet-Routing-Machine
+         * Can be used by the GetDistance or GetDuration methods and therefore will have specific attributes binded
+         * @param resolve Will be the resolve method from the Promise
+         * @param isDistance In case of a GetDistance this Boolean should be True. Otherwise, it should be False.
+         * @param e Event that got triggered (L.Routing.RoutingResultEvent) and respective structure
+         */
+        private _routesFoundHandler(
+            resolve?: OSFramework.Callbacks.Generic,
+            resolveType?: ResolveType,
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            e?: any
+        ) {
+            let bind: OSFramework.Callbacks.Generic;
+            // This method was invoked by the setRoute and because there might exist concurrency between the setRoute and the getDistance or getDirections methods
+            // when they are invoked one after the other (inside an action flow [OS] for instance), we need to ensure the listeners get removed after the "routesfound" event is triggered
+            this._currentDistance = e.routes[0].summary.totalDistance;
+            this._currentDuration = e.routes[0].summary.totalTime;
+            // If the resolve exists, means that this handler was invoked by the methods getDistance or getDuration as it has been passed via "bind"
+            // Otherwise, it means that the handler was invoked by the method setRoute on the creation of the route
+            if (resolve !== undefined) {
+                if (resolveType === ResolveType.GetDistance) {
+                    // Make sure to resolve with the callback "resolve" and passing the currentDistance as an argument
+                    resolve(this._currentDistance);
+                    // invoked by the getDistance method -> as soon as the routesFoundHandler gets executed and after getting the currentDistance and currentDuration we can later remove the listener for the getDistance.
+                    bind = this._bindDistance;
+                } else if (resolveType === ResolveType.GetDuration) {
+                    // Make sure to resolve with the callback "resolve" and passing the currentDuration as an argument
+                    resolve(this._currentDuration);
+                    // invoked by the getDuration method -> as soon as the routesFoundHandler gets executed and after getting the currentDistance and currentDuration we can later remove the listener for the getDuration.
+                    bind = this._bindDuration;
+                }
+            } else {
+                if (resolveType === ResolveType.SetRoute) {
+                    // invoked by the setRoute method -> as soon as the routesFoundHandler gets executed and after getting the currentDistance and currentDuration we can later remove the listener for the setRoute.
+                    bind = this._bindSetRoute;
+                }
```

Comment:
shouldn't we resolve the promise never the less?

Reply:
We don't have a promise for SetRoute. We just want to update the this.currentDistance and this.currentDuration variables